File: php/features.php
<?php
declare(strict_types=1);

// MediawikiAction SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MediawikiActionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MediawikiActionBaseFeature();
            case "test":
                return new MediawikiActionTestFeature();
            default:
                return new MediawikiActionBaseFeature();
        }
    }
}

File: py/features.py
# MediawikiAction SDK feature factory

from feature.base_feature import MediawikiActionBaseFeature
from feature.test_feature import MediawikiActionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MediawikiActionBaseFeature(),
        "test": lambda: MediawikiActionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
